crates/jit/src/link.rs

```diff
@@ -38,7 +38,11 @@ pub fn link_module(
                     signature,
                     vmctx,
                 } => {
-                    let import_signature = &module.signatures[module.functions[index]];
+                    let import_signature = {
+                        let sigidx = module.functions[index];
+                        let unique_index = module.signature_mapping[sigidx];
+                        &module.signatures[unique_index]
+                    };
```

Comment:
One minor point: given a function index, getting a signature is a little more convoluted. I've used a small block scope for the sake of keeping these clearly tied to getting `import_signature`.

Reply:
It looks like this sequence happens in a few different places. Could this code be factored into a helper method on `Module`?

Reply:
Yeah definitely! I'll take care of this.